From 08feaf64bd63422bd78cbc10b088ac337bde3d37 Mon Sep 17 00:00:00 2001
From: Potin Lai <potin.lai@quantatw.com>
Date: Fri, 22 May 2026 10:08:11 +0800
Subject: [PATCH] ARM: dts: aspeed: sanmiguel: Add virtual-mux-host node

NVIDIA uses a virtual-mux mechanism to allow the BMC to access
multiple FRU EEPROMs via a single address [1].

Add a virtual-mux-host node to enable BMC access to HPM and
HMC FRU EEPROMs.

[1] https://github.com/NVIDIA/linux/commit/4313ef5318b80ebdf7d8d74292b68a89e505fa6a

Signed-off-by: Potin Lai <potin.lai@quantatw.com>
---
 arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-sanmiguel.dts | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-sanmiguel.dts b/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-sanmiguel.dts
index f9f37a3deb30..4435ae6b2e5f 100644
--- a/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-sanmiguel.dts
+++ b/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-sanmiguel.dts
@@ -440,6 +440,12 @@ ssif-bmc@10 {
 
 &i2c1 {
 	status = "okay";
+
+	bridge@54 {
+		compatible = "virtual-mux-host";
+		reg = <0x54>;
+		virtual-bus-num = <90>;
+	};
 };
 
 &i2c2 {
-- 
2.53.0-Meta

